feat(Filter): implements new filter for water need #205

yannickoo opened the pull request:
> This PR adds the functionality to filter by the need of water.

## Demo

https://user-images.githubusercontent.com/1086961/124952034-31899500-e014-11eb-8645-7de9495ca1c1.mp4

## Steps to review
1. `npm start`
2. Open search
3. Choose "**Niedrig 💧**" as Wasserbedarf filter
4. Confirm that trees have been filtered
5. Choose "**Mittel 💧**" as Wasserbedarf filter
6. Confirm that trees have been filtered
7. Choose "**Hoch 💧**" as Wasserbedarf filter
8. Confirm that trees have been filtered
9. Click "**Hoch 💧**" filter again to deselect it
10. Confirm that trees are not filtered anymore

vogelino commented:
Hey @yannickoo! Thx a lot for this PR! Great work and I like your solution. We are currently in the middle of a sprint and we will look at your changes in detail after we get some time freed up. Concerning the update of the map, you might want to check the `updateTriggers` and the `componentDidUpdate` method. There, we check for updated props or state and redraw the map when necessary. Adding a comparison with your newly introduced state might do the trick.

yannickoo commented:
@vogelino I already updated `componentDidUpdate` but your hint with `updateTriggers` was great! I have updated the PR so now the map updates when changing water need filter 🎉

vogelino commented:
Hi @yannickoo! First and foremost: Thank you for your great contribution! It was well implemented, and you followed the way things were done. Much appreciated! We didn't have to change much in order to merge it. 

We did actually initially merge it but we had to roll it back as mobile wasn't covered by your implementation. For performance reasons, we use vector tiles on mobile, which also have to be filtered using the new state. I will have a look at what exactly is required to do that.

We are aware that the `DeckglMap` component is way too complex and needs to be simplified further (probably the reason why you missed this step). However, feel free to help us investigate how to do filtering on mobile. Looking at how this is achieved for other filters should provide a good entry point. 

Again, thanks a lot for your contribution! Cheers!
